app/src/main/java/com/tsu/wordsfactory/repository/WordsRepositoryImpl.kt

```diff
@@ -0,0 +1,80 @@
+package com.tsu.wordsfactory.repository
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import com.tsu.midtermexam3.data.WordModel
+import com.tsu.midtermexam3.data.WordsDao
+import com.tsu.wordsfactory.repository.data.DefinitionModel
+import com.tsu.wordsfactory.repository.data.MeaningModel
+import com.tsu.wordsfactory.repository.model.Definition
+import com.tsu.wordsfactory.repository.model.Meaning
+import com.tsu.wordsfactory.repository.model.Word
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import retrofit2.Response
+import java.util.*
+
+class WordsRepositoryImpl(private val wordsDatasource: WordsDatasource, private val wordsDao: WordsDao) : WordsRepository {
+
+    override suspend fun addWord(word: Word) {
```

Comment:
Better to separate this class to 2 classes. One is just return data, other proceed some business logic